src/services/api/contract_class.rs

```diff
@@ -84,12 +84,6 @@ impl ContractClass {
             return Err(ContractClassError::DisorderedBuiltins);
         };
 
-        if self.program.prime != *PRIME_STR {
```

Comment:
Instead of deleting this check that the `self.program.prime != *PRIME_STR` or `self.program.prime != "0c`

Reply:
Checking if `self.program.prime != 0` is wrong since any multiple of the prime would also be zero while not being the prime.

For example, let's say the prime is 13, then the prime (13) modulo prime would be zero, but so would 26, 39, 52...

Reply:
Nice catch @azteca1998 . we were trying to convert the prime into Felt and then to string, causing the value to be always zero. I fixed it moving the prime validation into the Program creation.